### PR TITLE
If the name doesn't match (nick name) use the e-mail #patch

### PR DESCRIPTION
**Pull request by charlottestjohn**
Did not put this into the user library we have; as trying to keep this module self sufficient for now

### CODE REVIEW

**Comment by ryanjfrizzell** [comment]
:shipit: LGTM

File: src/scripts/pagerduty.coffee [hunk 1]
```diff
@@ -170,13 +170,24 @@ module.exports = (robot) ->
     withCurrentOncallId: (msg, schedule, cb) ->
       robot.pagerduty.withCurrentOncallUser msg, schedule, (user, s) ->
         cb(user.id, user.name, s)
+
+    userForEmail: (email) ->
+      result = null
+      data = robot.brain.data.users
+      email = email.toLowerCase()
+      for k of (data or { })
```
Comment (partkyle) [inline src/scripts/pagerduty.coffee]: you can use this pattern to iterate over keys and values.

```
for key, value of (data) or {})
  userEmail = value['email_address']
  ...
```

**Comment by partkyle** [comment]
one comment, otherwise :shipit:

File: src/scripts/pagerduty.coffee (hunk 1 above)
Comment (charlottestjohn) [inline src/scripts/pagerduty.coffee]: So, I was trying to match the exact syntax of their userForName code: https://github.com/github/hubot/blob/master/src/brain.coffee#L121  but can switch to cleaner code if you'd like!

Comment (partkyle) [inline src/scripts/pagerduty.coffee]: Either is fine, the javascript code that is generated is identical.

**Pull request merged**